front/src/views/Boxes/components/transformers.ts

```diff
@@ -10,13 +10,13 @@ export const boxesRawDataToTableDataTransformer = (boxesQueryResult: BoxesForBox
       (element) =>
         ({
           labelIdentifier: element.labelIdentifier,
-          product: element.product!.name,
-          productCategory: element.product!.category.name,
-          gender: element.product!.gender,
+          product: element.product?.name,
```

Comment:
quick qn: Why are we removing the non-null assertion operator here? Is it possible that product/location can be optional?

Reply:
Good that you brought that up!

`!` aka [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) is a TypeScript compiler flag of sorts, telling TS to exclude `null` and `undefined` from the type of this property, or in a way telling it's okay if they end up being those types. 

But since this piece of code blew up for me using mock data, the right thing to do is to use `?` or the optional chaining operator to avoid exceptions on trying to access properties inside an `undefined` or `null` object/property.